python/ray/serve/controller.py

```diff
@@ -201,15 +205,25 @@ def autoscale(self) -> None:
                 if num_ongoing_requests is not None:
                     current_num_ongoing_requests.append(num_ongoing_requests)
 
-            if len(current_num_ongoing_requests) == 0:
-                continue
+            current_handle_queued_queries = self.handle_metrics_store.max(
+                deployment_name,
+                time.time() - autoscaling_policy.config.look_back_period_s,
+            )
+
+            if current_handle_queued_queries is None:
+                current_handle_queued_queries = 0
 
             new_deployment_config = deployment_config.copy()
 
             decision_num_replicas = autoscaling_policy.get_decision_num_replicas(
-                current_num_ongoing_requests=current_num_ongoing_requests,
                 curr_target_num_replicas=deployment_config.num_replicas,
+                current_num_ongoing_requests=current_num_ongoing_requests,
+                current_handle_queued_queries=current_handle_queued_queries,
             )
+
+            if decision_num_replicas == deployment_config.num_replicas:
```

Comment:
Is this for optimization?

Reply:
yes, we don't need to call deploy when num_replicas is not changed.